### PR TITLE
Builder for Yandex.Cloud

### PR DESCRIPTION
**Pull request by GennadySpb**
This PR adds the builder plugin to support [Yandex.Cloud](https://cloud.yandex.com/), trivial acceptance test and documentation included.

### CODE REVIEW

File: builder/yandex/driver_yc.go [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+
+func (d *driverYC) DeleteSubnet(ctx context.Context, subnetID string) error {
+	op, err := d.sdk.WrapOperation(d.sdk.VPC().Subnet().Delete(ctx, &vpc.DeleteSubnetRequest{
+		SubnetId: subnetID,
+	}))
+	if err != nil {
+		return err
+	}
+
+	err = op.Wait(ctx)
```
Comment (azr) [inline builder/yandex/driver_yc.go]: Nice one ! 👍

**Comment by azr** [comment]
Hello there, I reviewed the PR a little and it looks pretty good so far !

That said; the `Builder` interface recently changed to allow for context cancellation.
```suggestion
func (b *Builder) Run(ctx context.Context, ui packer.Ui, hook packer.Hook) (packer.Artifact, error) {
```

You will need to update your branch with master to get the latest changes, sorry this falls onto you, bad timing 🙂.

File: builder/yandex/builder.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	if errs != nil {
+		return warnings, errs
+	}
+	b.config = c
+	return warnings, nil
+}
+
+// Run executes a yandex Packer build and returns a packer.Artifact
+// representing a Yandex.Cloud compute image.
+func (b *Builder) Run(ui packer.Ui, hook packer.Hook) (packer.Artifact, error) {
```
Comment (azr) [inline builder/yandex/builder.go]:
```suggestion
func (b *Builder) Run(ctx context.Context, ui packer.Ui, hook packer.Hook) (packer.Artifact, error) {
```

File: builder/yandex/builder.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+	artifact := &Artifact{
+		image:  image.(*compute.Image),
+		config: b.config,
+	}
+	return artifact, nil
+}
+
+// Cancel.
+func (b *Builder) Cancel() {
```
Comment (azr) [inline builder/yandex/builder.go]: This Cancel func can be removed now with context support 🙂

**Comment by azr** [comment]
I just asked around, and I think we will need you to remove the `builder/yandex/LICENSE` and the `builder/yandex/THIRD-PARTY` files, the CLA you signed before states that the code you did is yours and that you accept that it goes under [packer's](https://github.com/hashicorp/packer/blob/master/LICENSE) license which is also a Mozilla Public License 2.0.
Otherwise I am afraid we cannot merge your work.

**Comment by GennadySpb** [comment]
> I just asked around, and I think we will need you to remove the `builder/yandex/LICENSE` and the `builder/yandex/THIRD-PARTY` files, the CLA you signed before states that the code you did is yours and that you accept that it goes under [packer's](https://github.com/hashicorp/packer/blob/master/LICENSE) license which is also a Mozilla Public License 2.0.
> Otherwise I am afraid we cannot merge your work.

Done.

**Comment by GennadySpb** [comment]
> You will need to update your branch with master to get the latest changes, sorry this falls onto you, bad timing 🙂.

No problem at all :-). Also fixed.

**Comment by GennadySpb** [comment]
> Nice one, LGTM, thanks for reacting so quickly ! 🙂

Thanks for the precise and fast review! 
What next steps?

**Comment by SwampDragons** [comment]
Can you add yourself to https://github.com/hashicorp/packer/blob/master/CODEOWNERS so github knows who to suggest for future reviews of this code?

**Comment by GennadySpb** [comment]
> Can you add yourself to https://github.com/hashicorp/packer/blob/master/CODEOWNERS so github knows who to suggest for future reviews of this code?

[Added](https://github.com/hashicorp/packer/pull/7484/commits/ae0d8b78e9f0b26c474492a521a9fca5b56d6669) myself and my colleagues.

**Comment by azr** [comment]
Awesome, thanks @GennadySpb !

**Comment by GennadySpb** [comment]
@azr Good news!  Thanks! 

When the next release is planned?

**Comment by azr** [comment]
We try to have a release cadence of ~6 weeks and last release was last week.

**Pull request merged**